File: CefSharp/Internals/JavascriptCallbackFactory.cs
using System;

namespace CefSharp.Internals
{
    public class JavascriptCallbackFactory : IJavascriptCallbackFactory
    {
        private readonly WeakReference browserProcessServiceHost;

        public JavascriptCallbackFactory(WeakReference browserProcessServiceHost)
        {
            this.browserProcessServiceHost = browserProcessServiceHost;
        }

        public IJavascriptCallback Create(JavascriptCallback callback)
        {
            return new JavascriptCallbackProxy(callback.Id, callback.BrowserId, browserProcessServiceHost);
        }
    }
}

File: CefSharp.Core/Internals/JavascriptCallbackFactory.cpp
// Copyright © 2010-2017 The CefSharp Authors. All rights reserved.
//
// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.

#include "Stdafx.h"
#include "JavascriptCallbackProxy.h"
#include "JavascriptCallbackFactory.h"

namespace CefSharp
{
    namespace Internals
    {
        IJavascriptCallback^ JavascriptCallbackFactory::Create(JavascriptCallback^ callback)
        {
            return gcnew JavascriptCallbackProxy(callback, _pendingTasks, BrowserAdapter);
        }
    }
}